packages/blitz-rpc/src/data-client/react-query-utils.ts

```diff
@@ -124,24 +124,26 @@ const sanitize =
 export const sanitizeQuery = sanitize("query")
 export const sanitizeMutation = sanitize("mutation")
 
-export const getQueryKeyFromUrlAndParams = (url: string, params: unknown) => {
-  const queryKey = [url]
-
-  const args = typeof params === "function" ? (params as Function)() : params
-  queryKey.push(serialize(args) as any)
+type BlitzQueryKey = [string] | [string, any]
+export const getQueryKeyFromUrlAndParams = (url: string, ...params: [unknown]): BlitzQueryKey => {
```

Comment:
The type says that it's a single-element array. Why are we spreading then?

Reply:
So that if it's optional it's an empty array, otherwise we'd get undefined

```ts
foo(1, 2);
foo(1, undefined);
foo(1);
```

If it's a single param first would be 2, the other 2 would be undefined
If it's an array param it's [2], [undefined], [] <- which is used to tell react-query to clear all keys

This could have also been solved by defaulting to something like "BLITZ_EMPTY_KEY_DONT_PASS" but that's kinda hacky 😅